eth/api.go

```diff
@@ -173,12 +173,20 @@ func (api *PrivateAdminAPI) ExportChain(file string) (bool, error) {
 	if err != nil {
 		return false, err
 	}
-	defer out.Close()
+	defer func() {
+		if err := out.Close(); err != nil {
+			panic(err)
+		}
+	}()
 
 	var writer io.Writer = out
 	if strings.HasSuffix(file, ".gz") {
 		writer = gzip.NewWriter(writer)
-		defer writer.(*gzip.Writer).Close()
+		defer func() {
+			if err := writer.(*gzip.Writer).Close(); err != nil {
+				panic(err)
```

Comment:
gzip writers flush buffered output on write. It'd be best to check for the error and ensure the error from this close call is returned from `ExportChain` when it happens. Maybe make the error return value named and assign it in the deferred function.